airflow/executors/fargate_executor.py

```diff
@@ -0,0 +1,336 @@
+            # get state of current task
+            task_state = task.get_task_state()
+            task_key = self.active_workers.arn_to_key[task.task_arn]
+            # mark finished tasks as either a success/failure
+            if task_state == State.FAILED:
+                self.fail(task_key)
+            elif task_state == State.SUCCESS:
+                self.success(task_key)
+            # if task is no longer running then remove from active workers and place in finished
+            if task_state != State.RUNNING:
+                self.log.debug(f'Task {task_key} marked as {task_state} after running on {task.task_arn}')
+                self.active_workers.pop_by_key(task_key)
+
+    def _sync_api_failures(self, arn_failures):
+        """
+        In the case of an API failure this executor will try a set amount of retries. Sometimes
+        one call to AWS might fail intermittently, and just needs to be called again in the next heartbeat. If
+        a given API call fails consecutively to a 'MAX_FAILURE_CHECKS' number of calls, the task is marked as failed.
+        """
+        for failure in arn_failures:
+            arn = failure["arn"]
+            task = self.active_workers.task_by_arn(arn)
+            task.api_failure_count += 1
+            # Sometimes ECS doesn't update the ARN key right away, and will show up as missing. Check later.
+            if task.api_failure_count >= self.__class__.MAX_FAILURE_CHECKS:
+                task_key = self.active_workers.arn_to_key[arn]
+                self.active_workers.pop_by_key(task_key)
+                self.log.error(f'ECS Executor could not find task {arn} because {failure["reason"]}; '
+                               f'marking key {task_key} as failed')
+                self.fail(task_key)
+            else:
+                self.log.debug(f'ECS Executor could not find task {arn} because {failure["reason"]}; '
+                               f'skipping for now on strike #{task.api_failure_count}')
+
+    def execute_async(self, key: TaskInstanceKeyType, command: CommandType,
+                      queue=None, executor_config=None):
+        """
+        This method will execute the command asynchronously.
+        """
+        # run a task
+        task_execution_api = self.executor_config_function(command)
+        try:
+            task_response = self.ecs.run_task(**task_execution_api)
+            self.log.debug(f'Executor responded with "{task_response}"')
+            if task_response['failures'] or not task_response['tasks']:
+                raise FargateError(', '.join([str(f) for f in task_response['failures']]))
+            task = FargateTask(task_response['tasks'][0])
+            self.log.info(f'Executor running task "{key}" on "{task.task_arn}"')
+            self.active_workers.add_task(task, key)
+        except (FargateError, Boto3Error) as e:
+            self.log.error(f'Executor failed to run task {key} with the following exception: {str(e)}')
```

Comment:
```suggestion
            self.log.error('Executor failed to run task %s with the following exception: %s.', key, str(e))
```
It's preferred to use built-in mechanism for formatting logs.

Reply:
Cool, I'll make the change for all f-strings

Reply:
I get f-strings are only 3.6 compatible., but now that python 2 is sun-setting is .format() cool yet?

Reply:
We can use f-strings freely, just not in logs where we prefer to use builtin formatting.